File: frontend/test/auth_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:bora_la/features/auth/data/auth_repository.dart';

void main() {
  test('login succeeds with email and password', () async {
    final repository = AuthRepository();

    final user = await repository.login('user@example.com', '123456');

    expect(user.email, 'user@example.com');
    expect(user.name, 'Usuário BoraLá');
  });

  test('login fails with empty credentials', () async {
    final repository = AuthRepository();

    expect(
      () async => repository.login('', ''),
      throwsA(isA<ArgumentError>()),
    );
  });
}

File: backend/test/bora_la/auth_test.exs
defmodule BoraLa.AuthTest do
  use ExUnit.Case, async: true

  alias BoraLa.Accounts.Auth

  test "login with valid credentials returns user" do
    assert {:ok, %{email: "user@example.com", name: "Usuário BoraLá"}} =
             Auth.login("user@example.com", "123456")
  end

  test "login with empty credentials returns error" do
    assert {:error, :invalid_credentials} = Auth.login("", "")
  end
end
